src/worker.rs

```diff
@@ -276,6 +296,44 @@ async fn load_older(client: &Client, store: &AsyncProgramStore) {
     }
 }
 
+fn insert_preview(
+    store: AsyncProgramStore,
+    room_id: OwnedRoomId,
+    source: PreviewSource,
+    media: Media,
+) {
```

Comment:
Maybe there should be just one task for the previews that we send the `PreviewSource`s to with a channel.